### PR TITLE
Various clean up

### PR DESCRIPTION
**Pull request by Log1x**
# Change log

- Clean up linting rules
- Handle rules in `phpcs` CLI instead of a separate `phpcs.xml`
- Fix & clean up various syntax/docblocks to be more compliant with PSR-12
- Add config optimizations to `composer.json`
- Bump dependencies
- Change build badge to CircleCI

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -45,6 +42,13 @@
     "squizlabs/php_codesniffer": "^3.4.2",
     "roave/security-advisories": "dev-master"
   },
+  "config": {
+    "optimize-autoloader": true,
+    "preferred-install": "dist",
+    "sort-packages": true
+  },
+  "minimum-stability": "dev",
```
Comment (austinpray) [inline composer.json]: What’s up with this?

File: composer.json [hunk 2]
```diff
@@ -58,7 +62,7 @@
       "php -r \"copy('.env.example', '.env');\""
     ],
     "test": [
-      "phpcs"
+      "phpcs --ignore=vendor,web --extensions=php --standard=PSR12 --exclude=PSR1.Files.SideEffects ."
```
Comment (austinpray) [inline composer.json]: Revert dis

File: config/application.php [hunk 3]
```diff
@@ -120,6 +130,6 @@
 /**
  * Bootstrap WordPress
  */
-if (!defined('ABSPATH')) {
-    define('ABSPATH', $webroot_dir . '/wp/');
+if (! defined('ABSPATH')) {
```
Comment (austinpray) [inline config/application.php]: Lol I hate this space after prefixed unary operators. Is this our coding style now?

File: phpcs.xml [hunk 4]
```diff
@@ -1,23 +0,0 @@
-<?xml version="1.0"?>
```
Comment (austinpray) [inline phpcs.xml]: Are we ditching this?

Comment (Log1x) [reply phpcs.xml]: only if I kept `phpcs --ignore=vendor,web --extensions=php --standard=PSR12 --exclude=PSR1.Files.SideEffects .` which p much does the same thing

File: config/application.php (hunk 3 above)
Comment (Log1x) [reply config/application.php]: https://www.php-fig.org/psr/psr-1/#23-side-effects scroll down little bit to conditional example. /shrug – seems to be the norm now

File: composer.json (hunk 1 above)
Comment (Log1x) [reply composer.json]: https://getcomposer.org/doc/articles/autoloader-optimization.md#optimization-level-1-class-map-generation + package sorting when doing `composer require ...` or even `composer update` after adding a package by hand (a-z)

**Comment by austinpray** [comment]
breaking this into multiple PRs

**Pull request closed without merging**